### PR TITLE
Public API: Acknowledge & Resolve actions

### PR DESCRIPTION
**Pull request by vadimkerr**
# What this PR does

Makes it possible to acknowledge/unacknowledge and resolve/unresolve alert groups via public API, and makes sure these actions are reflected properly in the alert group timeline.

## Demo

```bash
curl --request POST \
     --header "Authorization: TOKEN" \
     http://localhost:8080/api/v1/alert_groups/IQMHLV8INB24N/resolve
```

Screenshot 2023-10-04 at 16 05 27

## Which issue(s) this PR fixes

https://github.com/grafana/oncall/issues/3051

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: engine/apps/public_api/views/incidents.py [hunk 1]
```diff
@@ -112,3 +114,74 @@ def destroy(self, request, *args, **kwargs):
             wipe.apply_async((instance.pk, request.user.pk))
 
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+    @action(methods=["post"], detail=True)
```
Comment (vadimkerr) [inline engine/apps/public_api/views/incidents.py:118]: These API actions mostly duplicate the [internal API logic](https://github.com/grafana/oncall/blob/971384b50e2e4324fecd480a8abaa4cb8b7d78be/engine/apps/api/views/alert_group.py#L503-L554). I think these alert group state changes should be eventually refactored to be a proper FSM that can be reused in both APIs, but I want to leave this outside of the scope of this PR/issue.

Comment (matiasb) [reply engine/apps/public_api/views/incidents.py:118]: +1

File: engine/apps/alerts/models/alert_group_log_record.py [hunk 2]
```diff
@@ -258,13 +261,18 @@ def rendered_log_line_action(self, for_slack=False, html=False, substitute_autho
         elif self.escalation_policy is not None:
             escalation_policy_step = self.escalation_policy.step
 
-        if self.author is not None:
+        if self.action_source == ActionSource.API:
+            author_name = "API"
```
[the remark this answers is not in the document]
Comment (matiasb) [reply engine/apps/alerts/models/alert_group_log_record.py:265]: To be sure, we would still have the API token owner set as user in the record, right?

**Review by matiasb — APPROVED** [review]
LGTM

File: engine/apps/alerts/models/alert_group_log_record.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vadimkerr) [reply engine/apps/alerts/models/alert_group_log_record.py:265]: Yes, that's correct

**Pull request merged**